auto3dseg/algorithm_templates/dints/scripts/search.py

```diff
@@ -150,56 +159,42 @@ def run(config_file: Optional[Union[str, Sequence[str]]] = None, **override):
         )[dist.get_rank()]
     print("val_files:", len(val_files))
 
-    if torch.cuda.device_count() >= 4:
-        train_ds_a = monai.data.CacheDataset(
-            data=train_files_a,
-            transform=train_transforms,
-            cache_rate=1.0,
-            num_workers=8,
-            progress=False,
-        )
-        train_ds_w = monai.data.CacheDataset(
-            data=train_files_w,
-            transform=train_transforms,
-            cache_rate=1.0,
-            num_workers=8,
-            progress=False,
-        )
-        val_ds = monai.data.CacheDataset(
-            data=val_files,
-            transform=val_transforms,
-            cache_rate=1.0,
-            num_workers=2,
-            progress=False,
-        )
-    else:
-        train_ds_a = monai.data.CacheDataset(
-            data=train_files_a,
-            transform=train_transforms,
-            cache_rate=float(torch.cuda.device_count()) / 4.0,
-            num_workers=8,
-            progress=False,
-        )
-        train_ds_w = monai.data.CacheDataset(
-            data=train_files_w,
-            transform=train_transforms,
-            cache_rate=float(torch.cuda.device_count()) / 4.0,
-            num_workers=8,
-            progress=False,
-        )
-        val_ds = monai.data.CacheDataset(
-            data=val_files,
-            transform=val_transforms,
-            cache_rate=float(torch.cuda.device_count()) / 4.0,
-            num_workers=2,
-            progress=False,
-        )
+    train_cache_rate = float(parser.get_parsed_content("train_cache_rate"))
+    validate_cache_rate = float(parser.get_parsed_content("validate_cache_rate"))
+
+    train_ds_a = monai.data.CacheDataset(
+        data=train_files_a,
+        transform=train_transforms,
+        cache_rate=train_cache_rate,
+        num_workers=8,
+        progress=False,
+    )
+    train_ds_w = monai.data.CacheDataset(
+        data=train_files_w,
+        transform=train_transforms,
+        cache_rate=train_cache_rate,
+        num_workers=8,
+        progress=False,
+    )
+    val_ds = monai.data.CacheDataset(
+        data=val_files,
+        transform=val_transforms,
+        cache_rate=validate_cache_rate,
+        num_workers=2,
+        progress=False,
+    )
 
     train_loader_a = ThreadDataLoader(
-        train_ds_a, num_workers=6, batch_size=num_images_per_batch, shuffle=True
+        train_ds_a,
+        num_workers=parser.get_parsed_content("searching#num_workers"),
```

Comment:
Can we use `num_workers` for both searching and training in DiNTS?

Otherwise the user has to override the training_params with something like
```
train_params = {"num_workers": 4, "searching#num_workers": 4, "training#num_workers": 4}
```
in `AutoRunner`